### ISSUE TITLE
-[PFObject saveEventually] does not necessarily complete at a sensible time; misleading API

### ISSUE THREAD

**Issue by hokster7** [kastiglione/Parse-RACExtensions#26]
Since I thought `-saveEventually` is supposed to return more or less synchronously, I thought that the RAC extension `rac_saveEventually` would also complete within reasonable time.

This is not the case. `-rac_saveEventually` is implemented with `-saveEventually:`, whose parameter is a callback function.

The callback function is, in fact, never called if the save has not succeeded yet. Thus, this signal never actually completes. This is very odd behaviour and I think we should remove `-rac_saveEventually` from the API for now, since it represents a signal that could take a long time to complete.

**Comment by kastiglione** [comment]
Thanks for filing this. What do you think about documenting `-rac_saveEventually` to state that it may not complete, and that callers should consider using [`-timeout:onScheduler:`](https://github.com/ReactiveCocoa/ReactiveCocoa/blob/e6f5403d7b4eeb307740794758bc7ce19b773122/ReactiveCocoa/RACSignal%2BOperations.h#L542-L554)?

**Comment by hokster7** [comment]
@kastiglione yes I think that is a good idea — it's the solution I've gone with for now.

We must also be clear that timing out with `-timeout:onScheduler:` does not actually abort the `-saveEventually` operation happening on the background; it merely cuts the signal short. I think at present there's no way to 'cancel' a pending `-saveEventually`, so we're forced, by the Parse SDK, to be in a state of not knowing whether the save succeeded or not.
